### PR TITLE
Generate more meaningful signal name in lib.fsm

### PR DESCRIPTION
**Pull request by Konecho**
# Context, Motivation & Description
generate more meaningful & understandable signal name in fsm

# Impact on code generation

Only signal name changed, behaviour unchanged.

# Checklist

- [ ] Unit tests were added
- [ ] API changes are or will be documented:
  - using Scaladoc comments: `/** */`?
  - on [RTD](https://github.com/SpinalHDL/SpinalDoc-RTD)?
  - thanks to a [new tracking issue on RTD](https://github.com/SpinalHDL/SpinalDoc-RTD/issues/new?title=Document%20XXX&body=Do%20not%20merge%20until%20SpinalHDL/SpinalHDL%23XXX%20has%20not%20been%20merged)?

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/fsm/State.scala [hunk 1]
```diff
@@ -219,18 +219,18 @@ class StateMachineSharableRegUInt {
   * }}}
   *
   */
-class StateDelay(cyclesCount: UInt)(implicit stateMachineAccessor: StateMachineAccessor) extends State with StateCompletionTrait {
```
Comment (Readon) [inline lib/src/main/scala/spinal/lib/fsm/State.scala:222]: This modification seems to make the StateDelay's delay count unchangable.

Comment (Konecho) [reply lib/src/main/scala/spinal/lib/fsm/State.scala:222]: The motivation for this change is that spinal doesn't seem to have something like constant folding, causing the counter reset of the generated hdl to be split into two parts - the reset value comes at the very beginning of the file while the assigement at the end.

Comment (Konecho) [reply lib/src/main/scala/spinal/lib/fsm/State.scala:222]: I've changed the delay count api to be compatible with the previous `Int`, `UInt` or `TimeNumber` inputs.

Comment (Dolu1990) [reply lib/src/main/scala/spinal/lib/fsm/State.scala:222]: Ahhh interresting ^^ 
Seems ok to me now

**Comment by Dolu1990** [comment]
thanks ^^

**Pull request merged**